homeassistant/components/asuswrt/bridge.py

```diff
@@ -0,0 +1,289 @@
+"""aioasuswrt and pyasuswrt bridge classes."""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from collections import namedtuple
+import logging
+from typing import Any
+
+from aioasuswrt.asuswrt import AsusWrt as AsusWrtLegacy
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_MODE,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_PROTOCOL,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+from .const import (
+    CONF_DNSMASQ,
+    CONF_INTERFACE,
+    CONF_REQUIRE_IP,
+    CONF_SSH_KEY,
+    DEFAULT_DNSMASQ,
+    DEFAULT_INTERFACE,
+    KEY_METHOD,
+    KEY_SENSORS,
+    PROTOCOL_TELNET,
+    SENSORS_BYTES,
+    SENSORS_LOAD_AVG,
+    SENSORS_RATES,
+    SENSORS_TEMPERATURES,
+)
+
+SENSORS_TYPE_BYTES = "sensors_bytes"
+SENSORS_TYPE_COUNT = "sensors_count"
+SENSORS_TYPE_LOAD_AVG = "sensors_load_avg"
+SENSORS_TYPE_RATES = "sensors_rates"
+SENSORS_TYPE_TEMPERATURES = "sensors_temperatures"
+
+WrtDevice = namedtuple("WrtDevice", ["ip", "name", "connected_to"])
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _get_dict(keys: list, values: list) -> dict[str, Any]:
+    """Create a dict from a list of keys and values."""
+    ret_dict: dict[str, Any] = dict.fromkeys(keys)
+
+    for index, key in enumerate(ret_dict):
+        ret_dict[key] = values[index]
+
+    return ret_dict
```

Comment:
`dict(zip(keys, values))` will be more efficient:
```suggestion
    return dict(zip(keys, values))
```

Reply:
Done